packages/client/src/DisplayContent.tsx

```diff
@@ -90,29 +98,43 @@ function displayUnknown(
     subtype,
     content
   }: { type: string; subtype: string; content: string },
-  className: string
+  props: object
 ) {
   return content ? (
-    <div className={className}>
+    <div {...props}>
       <p>
         <em>
           [unknown content type: {type}/{subtype}]
         </em>
       </p>
     </div>
   ) : (
-    <div className={className}>
+    <div {...props}>
       <p>
         <em>[no content]</em>
       </p>
     </div>
   )
 }
 
-function handleExternalLink(event: React.MouseEvent<HTMLElement, MouseEvent>) {
+function handleLink(
+  accountId: string,
+  navigate: (location: string) => unknown,
+  event: React.MouseEvent<HTMLElement, MouseEvent>
+) {
   const target = event.target
   if (target instanceof HTMLAnchorElement && target.href) {
     event.preventDefault()
+
+    const parsed = parseMidUri(target.href)
+    const messageId = parsed && parsed.messageId
+    if (messageId) {
+      navigate(
+        `/accounts/${accountId}/conversations/${encodeURIComponent(messageId)}`
+      )
+      return
+    }
```

Comment:
If a link can be parsed as a `mid:` URI then we want to navigate to the corresponding conversation in the app. Otherwise we want to open the link in a browser using `openExternal`. For background on `mid:` URIs see https://tools.ietf.org/html/rfc2392